### PR TITLE
Fix unit tests

### PR DESCRIPTION
**Pull request by cdesiniotis**
This MR fixes some broken unit tests and bumps gpu-monitoring-tools so this project can be built on Mac.

Note, gpu-monitoring-tools is not maintained anymore and is strictly an archive.
Some follow-up work would be to migrate to using github.com/NVIDIA/go-nvml.

cc @rthallisey @zvonkok

### CODE REVIEW

**Comment by rthallisey** [comment]
Changes look fine to me.

File: pkg/device_plugin/generic_vgpu_device_plugin_test.go [hunk 1]
```diff
@@ -145,13 +145,15 @@ var _ = Describe("Generic Device", func() {
...
 	})
 
-	It("Should monitor health of device node", func() {
-		go dpi.healthCheck()
-		Expect(dpi.devs[0].Health).To(Equal(pluginapi.Healthy))
-		//time.Sleep(5 * time.Second)
-		unhealthy := <-dpi.unhealthy
-		Expect(unhealthy).To(Equal("1"))
-	})
+	/*
```
Comment (rthallisey) [inline pkg/device_plugin/generic_vgpu_device_plugin_test.go:148]: What's this test failing on?  Does it have something to do with a newer K8s?

Comment (cdesiniotis) [reply pkg/device_plugin/generic_vgpu_device_plugin_test.go:169]: The test would hang forever. The `healthCheck()` function would exit early with below:
```
2023/06/08 16:49:05 healthCheck(vGPUId): Unable to add device plugin socket path to fsnotify watcher: lstat /var/lib/kubelet/device-plugins: no such file or directory
```
I failed to realize this on the first go-around, and for whatever reason thought it was exiting early do to issues with initializing NVML.

I have updated the test file to create a dummy socket path and re-included this test case, which is now passing.

```
make test
go test ./... -coverprofile=coverage.out -v
=== RUN   TestDevicePlugin
Running Suite: DevicePlugin Suite
=================================
Random Seed: 1686272368
Will run 27 of 27 specs
. . .
Ran 27 of 27 Specs in 13.055 seconds
SUCCESS! -- 27 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestDevicePlugin (13.05s)
PASS
        kubevirt-gpu-device-plugin/pkg/device_plugin    coverage: 49.7% of statements
ok      kubevirt-gpu-device-plugin/pkg/device_plugin    13.522s coverage: 49.7% of statements
```

Comment (rthallisey) [reply pkg/device_plugin/generic_vgpu_device_plugin_test.go:169]: Great! Thanks

**Pull request merged**